Build fail on TFLite C API for Android with CMake

### Issue type

Build/Install

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.14.0-rc1

### Custom code

No

### OS platform and distribution

macOS 13.5.2

### Mobile device

Android

### Python version

_No response_

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

On version 2.14.0-rc1, Building TFLite C API for Android target with CMake fails by link error.
It works fine on version 2.13.0.

`ld: error: unable to find library -lpthread`
Error is caused by linking `pthread`. I guess pthread is unnecessary for android.

#### Extra information to my environment
- CMake: 3.26.4
- Android NDK: 25.2.9519653(r25c)

### Standalone code to reproduce the issue

```shell
mkdir tf-build
cd tf-build
cmake -DCMAKE_TOOLCHAIN_FILE=<NDK path>/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a ../tensorflow/lite/c
cmake --build . -j
```


### Relevant log output

```shell
...
[ 70%] Built target absl_status
[ 75%] Built target absl_flags
[ 95%] Built target tensorflow-lite
[100%] Linking CXX shared library libtensorflowlite_c.so
ld: error: unable to find library -lpthread
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libtensorflowlite_c.so] Error 1
make[1]: *** [CMakeFiles/tensorflowlite_c.dir/all] Error 2
make: *** [all] Error 2
```

--- a/tensorflow/lite/tools/cmake/modules/gemmlowp.cmake
+++ b/tensorflow/lite/tools/cmake/modules/gemmlowp.cmake
@@ -23,7 +23,7 @@ OverridableFetchContent_Declare(
   gemmlowp
   GIT_REPOSITORY https://github.com/google/gemmlowp
   # Sync with tensorflow/third_party/gemmlowp/workspace.bzl
-  GIT_TAG e844ffd17118c1e17d94e1ba4354c075a4577b88
+  GIT_TAG 16e8662c34917be0065110bfcd9cc27d30f52fdf
   # It's not currently (cmake 3.17) possible to shallow clone with a GIT TAG
   # as cmake attempts to git checkout the commit hash after the clone
   # which doesn't work as it's a shallow clone hence a different commit hash.

